GitCommands/Git/GitModule.cs

```diff
@@ -31,6 +32,7 @@ public sealed partial class GitModule : IGitModule
 
         public static readonly string NoNewLineAtTheEnd = "\\ No newline at end of file";
         public static CommandCache GitCommandCache { get; } = new();
+        public static CommandCache GitCommandCacheBetweenRefresh { get; } = new();
```

Comment:
To be frank, I'm finding the changes here rather worrisome. You're following the existing model, however, the model itself is broken in my opinion.

Could you please help me understand why we need this static API? Why do we also need `ClearGitCommandBetweenRefreshCache`?
 `IGitModule.ClearGitCommandBetweenRefreshCache` doesn't look as though it was thoroughly thought of... As in, is there `IGitModule.ClearGitCommand`? 

IIUIC, this API has no relation to the current git module (since it's a static). It also isn't something we'd expect to be used through the extensibility (i.e., plugins). And if my understanding is correct, why are we tying it to git module? 
If this cache has the same lifetime as the application, then why aren't we sticking it in the service container?


/cc: @gerhardol @mstv: this is precisely why I want the complete segregation of the interfaces and implementations so that we have design discussions and come up with API that both age gracefully with time, provide a solid foundation for extensibility and provide coherent and meaningful contract.